### ISSUE TITLE
Scenarios does not support assert diagnostics

### ISSUE THREAD

**Issue by v-jiaodi** [Azure/autorest.typescript#2924]
eg: https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/test/modularUnit/modelsGenerator.spec.ts#L2366
https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/test/modularUnit/operations.spec.ts#L90
https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/test/modularUnit/operations.spec.ts#L792
```
  it("should fail to handle model additional properties from spread record of int64 | string in non compatible mode", async () => {
    try {
      await emitModularModelsFromTypeSpec(
        `
      model Vegetables {
        ...Record<int64 | string>;
        carrots: int64;
        beans: int64;
      }
      op post(@body body: Vegetables): { @body body: Vegetables };
      `
      );
      assert.fail("Should throw diagnostic warnings");
    } catch (e) {
      const diagnostics = e as Diagnostic[];
      assert.equal(diagnostics.length, 1);
      assert.equal(
        diagnostics[0]?.code,
        "@azure-tools/typespec-ts/compatible-additional-properties"
      );
      assert.equal(diagnostics[0]?.severity, "warning");
    }
  });
```
